mypy/checker.py

```diff
@@ -507,8 +507,16 @@ def check_overlapping_overloads(self, defn: OverloadedFuncDef) -> None:
             # decorator or if the implementation is untyped -- we gave up on the types.
             inner_type = get_proper_type(inner_type)
             if inner_type is not None and not isinstance(inner_type, AnyType):
-                assert isinstance(inner_type, CallableType)
-                impl_type = inner_type
+                if isinstance(inner_type, CallableType):
+                    impl_type = inner_type
+                elif isinstance(inner_type, Instance):
+                    inner_call = get_proper_type(
+                        find_member('__call__', inner_type, inner_type, is_operator=True)
```

Comment:
`find_member` won't call any plugins. And might not as accurate as `analyze_member_access`. 

Any specific reason to use `find_member` here?

Reply:
I didn't have any specific reason, feel free to change! Just make sure that the getattr handling for the call operator matches runtime handling, e.g. with code like:
```
class X:
    def __getattr__(self, attr):
        if attr == "__call__":
            return lambda *a, **kw: print(a, kw)
        raise AttributeError

@X()
def f(): ...
```
In general, I'd love for member access to get cleaned up. I last tried a little bit in #8438 but that was swiftly reverted in #8500

Reply:
Related https://github.com/python/mypy/issues/3832

I will send a PR in a moment 🙂

Reply:
Right now it does not work this way with both `find_member` and `analyze_member_access`. I will have to add this.

Reply:
I think `find_member` does the right thing when passed `is_operator=True`? This is what I meant: https://github.com/python/mypy/pull/11637